File: language-support/js/daml-ledger/src/validation/completion_end_response.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from "..";
import { object, Validation } from "./base";
import { RequiredValidation } from "./base/typelevel";
import { LedgerOffset } from "./ledger_offset";

function required(): RequiredValidation<ledger.CompletionEndResponse> {
    return {
        offset: LedgerOffset
    };
}

export const CompletionEndResponse: Validation = object<ledger.CompletionEndResponse>('CompletionEndResponse', required, () => ({}));

File: language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/CompletionEndResponse.java
// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.javaapi.data;

import com.daml.ledger.api.v1.CommandCompletionServiceOuterClass;

import java.util.Objects;

public final class CompletionEndResponse {

    private final LedgerOffset offset;

    public CompletionEndResponse(LedgerOffset offset) {
        this.offset = offset;
    }

    public static CompletionEndResponse fromProto(CommandCompletionServiceOuterClass.CompletionEndResponse response) {
        return new CompletionEndResponse(LedgerOffset.fromProto(response.getOffset()));
    }

    public LedgerOffset getOffset() {
        return offset;
    }

    @Override
    public String toString() {
        return "CompletionEndResponse{" +
                "offset=" + offset +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        CompletionEndResponse that = (CompletionEndResponse) o;
        return Objects.equals(offset, that.offset);
    }

    @Override
    public int hashCode() {

        return Objects.hash(offset);
    }
}
